discovery-provider/src/model_validator.py

```diff
@@ -0,0 +1,160 @@
+import os
```

Comment:
@vicky-g is this like a test script?

Reply:
i eventually plan to convert this `ModelValidator` class into something we actually use. what is in this current PR is a test script